webrtc/sctp/sctp_utils.nim

```diff
@@ -0,0 +1,82 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import nativesockets
+import binary_serialization, chronos
+
+var errno* {.importc, header: "<errno.h>".}: cint ## error variable
+
+when defined(windows):
+  import winlean
+  const
+    SctpAF_INET* = winlean.AF_INET
+    SctpEINPROGRESS* = winlean.WSAEINPROGRESS.cint
+else:
+  const
+    SctpAF_INET* = nativesockets.AF_INET
+    SctpEINPROGRESS* = chronos.EINPROGRESS.cint
+
+type
+  # These three objects are used for debugging/trace only
+  SctpChunk* = object
+    chunkType*: uint8
+    flag*: uint8
+    length* {.bin_value: it.data.len() + 4.}: uint16
+    data* {.bin_len: it.length - 4.}: seq[byte]
+
+  SctpPacketHeader* = object
+    srcPort*: uint16
+    dstPort*: uint16
+    verifTag*: uint32
+    checksum*: uint32
+
+  SctpPacketStructure* = object
+    header*: SctpPacketHeader
+    chunks*: seq[SctpChunk]
```

Comment:
Could you add more info about these and why they are necessary?

Reply:
Done.